src/app/shared/case-add/case-add.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, Input, OnInit } from '@angular/core';
+import * as data from './case-add-form-data';
+import { CaseService } from '../../_services/api/case.service';
+import { FormBase } from '../dynamic-form/types/form-element-base';
+import { CaseDataDto } from '../../_models/caseDataDto';
+import { FormElementControlService } from '../../_services/form-element-control.service';
+
+@Component({
+  selector: 'app-case-add',
+  templateUrl: './case-add.component.html',
+  styleUrls: ['./case-add.component.scss'],
+})
+export class CaseAddComponent implements OnInit {
+  @Input() selectedResource: CaseDataDto;
+  myFormElements: FormBase<any>[] = [];
+
+  constructor(
+    public caseService: CaseService,
+    private formElementControlService: FormElementControlService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.selectedResource) {
+      this.myFormElements = this.formElementControlService.setValuesForDynamicForm(
+        this.selectedResource,
+        JSON.parse(JSON.stringify(data.FORM_DATA_CASE_ADD))
+      );
+    } else {
+      this.myFormElements = JSON.parse(JSON.stringify(data.FORM_DATA_CASE_ADD));
```

Comment:
Maybe we could put this in a constant before the if, and use that one in each case of the if.

Reply:
that's exactly what we need to avoid here